### PR TITLE
Karma quest sbv admin content request  all request Functionality implemented

### PR DESCRIPTION
**Pull request by sohailamjad12**

### CODE REVIEW

File: project/ws/app/src/lib/routes/home/routes/request/all-request/all-request.component.html [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                    <th mat-header-cell *matHeaderCellDef>Interests</th>
+                    <td mat-cell class="title-class" *matCellDef="let element">
+                        <a class="action-btn-view"  (click)="showInterestPopup()"
+                        href="javascript:void(0)">{{element?.interestCount}}</a>
+                    </td>
+                  </ng-container>
+      
+                  <ng-container matColumnDef="details">
+                    <th mat-header-cell *matHeaderCellDef>Details</th>
+                    <td mat-cell *matCellDef="let element">
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/routes/request/all-request/all-request.component.html:64]: please use optional check

File: project/ws/app/src/lib/routes/home/routes/request/all-request/all-request.component.html [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+                          <span>
+                        <button mat-icon-button [matMenuTriggerFor]="menu" aria-label="Example icon-button with a menu">
+                          <mat-icon>more_vert</mat-icon>
+                        </button>
+                        <mat-menu #menu="matMenu">
+                          <button mat-menu-item (click)="onClickMenu(element,'viewContent')">
+                            <span >View</span>
+                          </button>
+                          <button *ngIf="element.status === 'Unassigned' && element.status!== 'Inprogress' &&  element.status!== 'invalid' " mat-menu-item (click)="onClickMenu(element,'invalidContent')">
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/routes/request/all-request/all-request.component.html]: please validate hardcoded value by putting into variable and then use here

File: project/ws/app/src/lib/routes/home/routes/request/all-request/all-request.component.ts [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  isAssigned = false
+  inProgress = false
+  invalid = false
+  dataSource: any
+  displayedColumns: string[] = ['RequestId', 'title', 'requestType', 'requestStatus', 'assignee', 'requestedOn', 'interests', 'action']
+  dialogRef: any
+  queryParams: any
+  statusCards = [
+    {
+      title: 'Unassigned',
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/routes/request/all-request/all-request.component.ts:39]: are we using hardcoded values?

File: project/ws/app/src/lib/routes/home/routes/request/competency-view/competency-view.component.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    public dialogRef: MatDialogRef<CompetencyViewComponent>,
+    private sanitized: DomSanitizer,
+    @Inject(MAT_DIALOG_DATA) public dData: any
+  ) { }
+  ngOnInit() {
+    if (this.dData && this.dData.children && this.dData.children.length > 0) {
+      if (this.dData.selectedLevelId) {
+        this.levelSelected = this.dData.children.filter((v: any) => v.id === this.dData.selectedLevelId)[0]
+      } else {
+        this.levelSelected = this.dData.children[0]
```
Comment (vishnubansaltarento) [inline project/ws/app/src/lib/routes/home/routes/request/competency-view/competency-view.component.ts:23]: please validate children length, then assign 0

Comment (sohailamjad12) [reply project/ws/app/src/lib/routes/home/routes/request/competency-view/competency-view.component.ts:23]: already we are checking children length in line 19

File: project/ws/app/src/lib/routes/home/routes/request/all-request/all-request.component.ts (hunk 3 above)
Comment (sohailamjad12) [reply project/ws/app/src/lib/routes/home/routes/request/all-request/all-request.component.ts:39]: added enum fo status

File: project/ws/app/src/lib/routes/home/routes/request/all-request/all-request.component.html (hunk 1 above)
Comment (sohailamjad12) [reply project/ws/app/src/lib/routes/home/routes/request/all-request/all-request.component.html:64]: already added optional check

**Pull request merged**